### PR TITLE
Proof for aws_hash_table_get_entry_count

### PR DESCRIPTION
**Pull request by danielsn**
1. Add proof for ```aws_hash_table_get_entry_count()```
1. update hash table invariants to handle entry-count correctly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+void aws_hash_table_get_entry_count_harness() {
+    struct aws_hash_table table;
+    // There are no loops in the code under test, so use the biggest possible value
+    ensure_allocated_hash_table(&table, SIZE_MAX);
+    __CPROVER_assume(aws_hash_table_is_valid(&table));
+
+    struct store_byte_from_buffer stored_byte;
+    save_byte_from_hash_table(&table, &stored_byte);
+
+    aws_hash_table_get_entry_count(&table);
```
Comment (feliperodri) [inline .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c]: Could you previously store the `entry_count` and then check whether the function will always return the correct value?

File: .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#include <aws/common/hash_table.h>
+#include <aws/common/math.h>
+#include <aws/common/private/hash_table_impl.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+void aws_hash_table_get_entry_count_harness() {
+    struct aws_hash_table table;
+    // There are no loops in the code under test, so use the biggest possible value
```
Comment (JonathanHenson) [inline .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c]: no c++ comments pls

File: .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c (hunk 1 above)
Comment (danielsn) [reply .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c]: Done

File: .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c (hunk 2 above)
Comment (danielsn) [reply .cbmc-batch/jobs/aws_hash_table_get_entry_count/aws_hash_table_get_entry_count_harness.c]: Done

**Pull request merged**